src/mcp/server/lowlevel/func_inspection.py

```diff
@@ -0,0 +1,48 @@
+import inspect
+from collections.abc import Callable
+from typing import Any
+
+
+def accepts_request(func: Callable[..., Any]) -> bool:
```

Comment:
I think maybe the function we want is `def accepts_single_positional_arg(func, arg_type)`? (or `can_be_called_with_single_positional` to allow defaults)

Reply:
I think a handler w/ an untyped single pos param (or untyped `*args`) is perfectly fine, but if that single pos param does have a type it's a low hanging fruit to check it.